main.tf

```diff
@@ -17,12 +17,8 @@ provider "aws" {
 
 }
 
-data "aws_eks_cluster" "cluster" {
-  name = module.eks.cluster_id
-}
-
 data "aws_eks_cluster_auth" "cluster" {
-  name = module.eks.cluster_id
```

Comment:
Note for reviewers, `cluster_id` was deprecated where it no longer blocks until terraform is able to communicate with the cluster and no longer returns a value unless you are using AWS Outposts. The recommendation was to directly get those values from module.eks rather than using the aws_eks_cluster data object here which allows for the "block" to reintroduced.
See L61 and in locals.